mousestyles/density_estimation.py

```diff
@@ -0,0 +1,162 @@
+    compute the euclidian distance between the positions
+    of the mouse at two consecutive times.
+
+    More specifically:
+
+    - let delta_t be the step parameter.
+    - let t_n be the sequance of non negative numbers such
+    that t_0 = 0 and t_(n+1) = t_n + delta_t. The sequence
+    is defined for all n such that n>=0 and t_n <= time
+    of the experiment
+    - let d_n be the sequence of non negative numbers such
+    that d_0 = 0 and d_n equals the position of the mouse
+    at a particular day at time t_n. d_n is then defined
+    on the same set of integers as the sequence t_n.
+    - The function returns the sequence d_n.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    day: int
+        nonnegative integer indicating the day number
+    step: float
+        positive float defining the time between two observations
+
+    Returns
+    -------
+    movement : numpy array
+
+    Examples
+    --------
+    >>> dist = extract_distances(0, 0, 0, step=1e2)
+    0
+    """
+    movement = load_movement(strain, mouse, day)
+    # Compute distance between samples
+    dist = np.empty((movement.shape[0]-1, 2))
+    x = np.array(movement["x"])
+    y = np.array(movement["y"])
+    # dist[:, 0] contains the distances between two
+    # consecutive points
+    dist[:, 0] = np.sqrt((x[1:] - x[:-1])**2 + (y[1:] - y[:-1])**2)
+    # dist[:, 1] contains the recorded times for which
+    # the distances have been computed
+    dist[:, 1] = np.array(movement['t'])[1:]
+    dist[:, 1] = dist[:, 1] - dist[0, 1]
+    tf = dist[dist.shape[0] - 1, 1]
+    # Aggregate distances according to step
+    aggregate = np.zeros(int(tf/step))
+    j = 0
+    for i in range(len(aggregate)):
+        while dist[j, 1] < i*step:
+            aggregate[i] = aggregate[i] + dist[j, 0]
+            j = j+1
+    return(aggregate)
+
+
+def extract_distances_bymouse(strain, mouse, step=1e2, verbose=False):
+    """
+    Aggregates extract_distances for all days of recorded data for
+    one particular mouse. Then returns the mean of the aggregated data.
+
+    More specifically:
+
+    - let d^1,...,d^D be the sequence of distances for one particular
+    mouse for days 1 to D.
+    - let d^mouse the empty sequence of length the maximal length of
+    d^1,...,d^D, defined as follow:
+    for each index i in d^mouse, d^mouse[i] is equal to the mean
+    over j of the existing d^j[i].
+    - The function returns the sequence d^mouse.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    step: float
+        positive float defining the time between two observations
+
+    Returns
+    -------
+    movement : numpy array
+
+    Examples
+    --------
+    >>> dist = extract_distances_bymouse(0, 0)
```

Comment:
I think the example should be like
>>> extrace_distances_bymouse(0, 0)
[1,2,3]
...
[1,2,3] # Whatever the result of that statement here